umh.docs.umh.app/content/en/docs/getstarted/dataacquisitionmanipulation.md

```diff
@@ -1,205 +1,289 @@
-  {{% /notice %}}
+Enter your Kafka broker's details:
 
-6. In this column, you can view all incoming messages and their respective
-   topics. The incoming topics follow this format:
-   `ia/raw/development/ioTSensors/`. To keep it simple, this tutorial will be
-   only using the temperature topic, but feel free to choose any
-   topic you'd like. Copy the temperature topic
-   `ia/raw/development/ioTSensors/Temperature`, open the **mqtt-in** node,
-   paste the copied topic in the **Topic** field, click on **Done**, and then
-   press **Deploy** again to apply the changes.
+- **Server**: united-manufacturing-hub-kafka
+- **Port**: 9092
 
-   ![Untitled](/images/getstarted/dataAcquisitionManipulation/getStartedDataAcqManNewTopic.png)
+Click **Add** to save.
 
-7. To format the incoming message according to the UMH data-model, add a
-   **function** node, an **mqtt-out** and an **mqtt-in** to your flow. 
-   Connect the nodes in the following order:
-   **mqtt-in → function → mqtt-out** and the second flow: **mqtt-in → debug**.
+![Connect Kafka to Node-RED](/images/getstarted/dataAcquisitionManipulation/noderedKafkaConnection.png?width=80%)
 
-   ![Untitled](/images/getstarted/dataAcquisitionManipulation/getStartedDataAcqManNewNodes.png)
+Set the subscription topic. For demonstration, we'll use the topic created earlier:
 
-{{% notice info %}}
-Incoming messages are not following the UMH data-model. To contextualize and
-format them correctly, you can use the different message types. This is needed,
-to ensure that the data is correctly processed by the UMH stack and other 
-consumers.
+```text
+umh.v1.pharma-genix.aachen.packaging.packaging_1.blister
+```
 
-In the documentation, you can find the complete
-[list](https://umh.docs.umh.app/docs/architecture/datamodel/messages/)
-of all available message types and the respective topics.
+Link a debug node to the kafka-consumer node, deploy, and observe messages in the
+debug pane.
+
+![Kafka Debug Connection](/images/getstarted/dataAcquisitionManipulation/noderedKafkaDebugConnection.png?width=30%)
+
+{{% notice note %}}
+For topic structuring guidelines, refer to [Unified Namespace](/docs/features/unified-namespace).
+{{% /notice %}}
+
+### Format Incoming Messages
+
+Since the data is already processed from the previous step, use a function node
+to convert the temperature from Celsius to Fahrenheit. Connect it to the
+kafka-consumer node and paste the following script:
+
+```jsx
+if (msg.payload.key != "temperatureCelsius") {
+    return null;
+}
+var celsius = Number(msg.payload.value)
+var fahrenheit = (celsius * 9 / 5) + 32
+
+msg.payload = String(fahrenheit)
+msg.key = "temperatureFahrenheit" // Save using a different key
+
+return msg;
+```
+
+This script selectively processes messages with the temperatureCelsius key,
+converting the value to Fahrenheit.
+
+Finalize with **Done**.
+
+### Send Formatted Data Back to Kafka
```

Comment:
double check, i remember there was some weird stuff happening here

(you get the payload as payload, but need to send it as payload.value or something)